packages/cli/src/commands/redeploy.ts

```diff
@@ -0,0 +1,214 @@
+import { isErrnoException } from '@vercel/error-utils';
+import logo from '../util/output/logo';
+import Now from '../util';
+import { printDeploymentStatus } from '../util/deploy/print-deployment-status';
+import stamp from '../util/output/stamp';
+import ua from '../util/ua';
+import type { VercelClientOptions } from '@vercel/client';
+
+const help = () => {
+  console.log(`
+  ${chalk.bold(
+    `${logo} ${getPkgName()} redeploy`
+  )} [deploymentId|deploymentName]
+
+  Rebuild and deploy a previous deployment.
+
+  ${chalk.dim('Options:')}
+
+    -h, --help                     Output usage information
+    -A ${chalk.bold.underline('FILE')}, --local-config=${chalk.bold.underline(
+    'FILE'
+  )}   Path to the local ${'`vercel.json`'} file
+    -Q ${chalk.bold.underline('DIR')}, --global-config=${chalk.bold.underline(
+    'DIR'
+  )}    Path to the global ${'`.vercel`'} directory
+    -d, --debug                    Debug mode [off]
+    --no-color                     No color mode [off]
+    --no-wait                      Don't wait for the redeploy to finish
+    -t ${chalk.bold.underline('TOKEN')}, --token=${chalk.bold.underline(
+    'TOKEN'
+  )}        Login token
+    -y, --yes                      Skip questions when setting up new project using default scope and settings
+
+  ${chalk.dim('Examples:')}
+
+  ${chalk.gray('–')} Rebuild and deploy an existing deployment using id or url
+
+    ${chalk.cyan(`$ ${getPkgName()} redeploy <deployment id/url>`)}
+
+  ${chalk.gray('–')} Redeploy a deployment using id or url
+
+    ${chalk.cyan(`$ ${getPkgName()} redeploy <deployment id/url>`)}
+
+  ${chalk.gray('–')} Write Deployment URL to a file
+
+    ${chalk.cyan(
+      `$ ${getPkgName()} redeploy my-deployment.vercel.app > deployment-url.txt`
+    )}
+`);
+};
+
+/**
+ * `vc redeploy` command
+ * @param {Client} client
+ * @returns {Promise<number>} Resolves an exit code; 0 on success
+ */
+export default async (client: Client): Promise<number> => {
+  let argv;
+  try {
+    argv = getArgs(client.argv.slice(2), {
+      '--debug': Boolean,
+      '-d': '--debug',
+      '--no-wait': Boolean,
+      '--yes': Boolean,
+      '-y': '--yes',
+    });
+  } catch (err) {
+    handleError(err);
+    return 1;
+  }
+
+  if (argv['--help'] || argv._[0] === 'help') {
+    help();
+    return 2;
+  }
+
+  const { output } = client;
+  const deployId = argv._[1];
+  if (!deployId) {
+    output.error(
+      `Missing required deployment id or url: ${getCommandName(
+        `redeploy <deployment-id-or-url>`
+      )}`
+    );
+    return 1;
+  }
+
+  const { contextName } = await getScope(client);
+  const noWait = !!argv['--no-wait'];
+
+  try {
+    const fromDeployment = await getDeploymentByIdOrURL({
+      client,
+      contextName,
+      deployId,
+    });
+
+    const deployStamp = stamp();
+    output.spinner(`Redeploying project ${fromDeployment.id}`, 0);
+
+    let deployment = await client.fetch<any>(`/v13/deployments?forceNew=1`, {
```

Comment:
Can we get rid of `any`?

Reply:
In short, yes, but it's outside the scope of this PR. This line was copied from `deploy.ts`. Additionally, `deployment` here is passed around into other functions such as `checkDeploymentStatus()` in `client` which uses a different `Deployment` type than the one used in `cli` and `@vercel-internals/types`. There should not be multiple versions of the `Deployment` type, but fixing it is going to touch a lot of code, so best to do it in a different PR.